### PR TITLE
feat(eslint-plugin): backport no-unsafe-function type, no-wrapper-object-types from v8 to v7

### LINKED ISSUE #9503: Docs: Mention deprecation and replacement rules in ban-types docs
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Right now, `ban-types` has very different docs pages on...
* `main`: https://typescript-eslint.io/rules/ban-types, with a happy ✅ indicating it's recommended
* `v8`: https://v8--typescript-eslint.netlify.app/rules/ban-types, with a stern red deprecation notice

Such whiplash to jump between the two! Let's add a big Markdown admonition on top of `ban-types.mdx` in the `main` branch saying it's deprecated and will be removed in v8. 

### Affected URL(s)

https://typescript-eslint.io/rules/ban-types

💖

### LINKED ISSUE #9505: eslint-plugin: Also backport no-unsafe-function-type and no-wrapper-object-types from v8 to main
## Overview

Continuing #9437 -> #9443: two additional rules were split out of `ban-types` that we'll want to backport from the `v8` branch to `main`:

* https://v8--typescript-eslint.netlify.app/rules/no-unsafe-function-type
* https://v8--typescript-eslint.netlify.app/rules/no-wrapper-object-types

I'll take this on so we can get it in the weekly release in two days.

cc @Vinccool96 as FYI: your PR was excellent, so I'm going to do the same thing here. 🙂 

💖

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9503; fixes #9505
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Also adds the docs touchup requested by #9503 while I'm in the area.

💖

### CODE REVIEW

**Comment by kirkwaiblinger** [comment]
FYI @JoshuaKGoldberg - I've merged https://github.com/typescript-eslint/typescript-eslint/pull/9363 to v8, so you'll want to update this PR with those changes

**Review by kirkwaiblinger — CHANGES_REQUESTED** [review]
https://github.com/typescript-eslint/typescript-eslint/pull/9507#issuecomment-2224835625

File: packages/eslint-plugin/docs/rules/no-wrapper-object-types.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+You might consider using [ESLint disable comments](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) for those specific situations instead of completely disabling this rule.
+
+## Further Reading
+
+- [MDN documentation on primitives](https://developer.mozilla.org/en-US/docs/Glossary/Primitive)
+- [MDN documentation on `string` primitives and `String` objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_primitives_and_string_objects)
+
+## Related To
+
+- [`no-empty-object-type`](https://v8--typescript-eslint.netlify.app/rules/no-empty-object-type)
```
Comment (kirkwaiblinger) [inline packages/eslint-plugin/docs/rules/no-wrapper-object-types.mdx:73]: Is the v8 link still necessary?
```suggestion
- [`no-empty-object-type`](./no-empty-object-type.mdx)
```

File: packages/eslint-plugin/docs/rules/no-unsafe-function-type.mdx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+</Tabs>
+
+## When Not To Use It
+
+If your project is still onboarding to TypeScript, it might be difficult to fully replace all unsafe `Function` types with more precise function types.
+You might consider using [ESLint disable comments](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) for those specific situations instead of completely disabling this rule.
+
+## Related To
+
+- typescript-eslint@v8's [`no-empty-object-type`](https://v8--typescript-eslint.netlify.app/rules/no-empty-object-type)
```
Comment (kirkwaiblinger) [inline packages/eslint-plugin/docs/rules/no-unsafe-function-type.mdx:61]: can these also be updated?
```suggestion
- [`no-empty-object-type`](https://v8--typescript-eslint.netlify.app/rules/no-empty-object-type)
```

**Review by bradzacher — APPROVED** [review]

**Review by kirkwaiblinger — DISMISSED** [review]

**Review by kirkwaiblinger — APPROVED** [review]

**Comment by JoshuaKGoldberg** [comment]
Aha! The check from #9243 is saving us on this one.

**Review by kirkwaiblinger — APPROVED** [review]

**Review by kirkwaiblinger — CHANGES_REQUESTED** [review]
Looks like might need to rerun some test snapshots? Otherwise LGTM

**Review by kirkwaiblinger — APPROVED** [review]
🚀

**Pull request merged**